proof/archive.go

```diff
@@ -268,40 +269,48 @@ func (m *MultiArchiver) ImportProofs(ctx context.Context,
 	// Before we import the proofs into the archive, we want to make sure
 	// that they're all valid. Along the way, we may augment the locator
 	// for each proof accordingly.
-	//
-	// TODO(roasbeef): can do concurrently
+	eg := &errgroup.Group{}
 	for _, proof := range proofs {
 		proof := proof
+		eg.Go(func() error {
+			// First, we'll decode and then also verify the proof.
+			finalStateTransition, err := m.proofVerifier.Verify(
+				ctx, bytes.NewReader(proof.Blob),
+			)
+			if err != nil {
+				return fmt.Errorf("unable to verify proof: %w", err)
```

Comment:
nit: maybe we should include the idx or something in the err that allows us identify which proof is failing

Reply:
to be honest I don't see any value that can be identifier for proof in the error.

Maybe `Locator.ScriptKey`, but it can be empty too :thinking:

Reply:
maybe the outpoint?

That's something that I was not sure if it should be included in the generic function or not

 `ErrGroup[V any, S []V](f func(int, V) error, s S)` where f looks like `func (idx int, v V) error` so it gets the index of the for/identifier of the task

Reply:
As I see now `proof.MultiArchiver` is used only in `tarogarden/custodian` file and there is specified `Locator.AssetID`, also this function is passing only 1 proof :D

I can only propose to add the first n bytes of raw proof if it makes sense :thinking: